config/env_config.go

```diff
@@ -339,3 +339,30 @@ func setBoolPtrFromEnvVal(dst **bool, keys []string) error {
 
 	return nil
 }
+
+func setEndpointDiscoveryTypeFromEnvVal(dst **EndpointDiscoveryEnableState, keys []string) error {
+	for _, k := range keys {
+		value := os.Getenv(k)
+		if len(value) == 0 {
+			continue
+		}
+
+		if *dst == nil {
+			*dst = new(EndpointDiscoveryEnableState)
+		}
+
+		switch {
+		case strings.EqualFold(value, "false"):
+			**dst = EndpointDiscoveryDisabled
+		case strings.EqualFold(value, "true"):
+			**dst = EndpointDiscoveryEnabled
+		case strings.EqualFold(value, "auto"):
+			**dst = EndpointDiscoveryAuto
```

Comment:
These comparisons should be against the constant values of `false`, `true`, `auto`.